docs/architecture/adr-048-consensus-fees.md

```diff
@@ -0,0 +1,181 @@
+# ADR 048: Consensus Fees
+
+## Changelog
+
+- Dec 1, 2021: Initial Draft
+
+## Status
+
+Proposed
+
+## Abstract
+
+This ADR describes a machanism to maintain the minimal gas price on consensus level.
+
+## Context
+
+Currently, each validator configures it's own `minimal-gas-prices` in `app.yaml`. But setting a proper minimal gas price is critical to protect network from dos attack, and it's hard for all the validators to pick a sensible value, so we proposal to maintain the minimal gas price in consensus level.
+
+Since tendermint 0.35 has supported mempool prioritization, we can take advantage of that to implement more sophisticated gas fee system.
+
+### Compare Different Designs
+
+#### Constant gas fee
+
+##### Positive
+
+- Predictable.
+
+  This makes client behaviour simpler and predictable. And with current system, users are already live with a de facto constant gas price since most validators don't adjust their minimal gas prices.
+
+##### Negative
+
+- Can't include urgent transactions with priority.
+- Don't adjusted based on the block load.
+
+#### First price auction
+
+Before EIP-1559, ethereum miners select transactions ranked by the highest fees, which creates a first price auction market.
+
+##### Positive
+
+- Do provide prioritized service based on tx gas price.
+
+##### Negative
+
+- Unpredictable.
+
+- Overpricing.
+
+#### EIP-1559
+
+[EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) combines a deterministic in-protocol base price and an auctioned tip price.
+
+##### Positive
+
+- Can prioritize urgent transactions.
+- More deterministic than the previous pure auction system.
+
+##### Negative
+
+- Unpredictable.
+- Overpricing the tip price in theory TODO I guess we don't obverse this in real world on ethereum ?
```

Comment:
if there is a sudden congestion owing to huge price volatility of assets, EIP-1559 will degrade to first price auction